api/v1alpha1/tcproute_types.go

```diff
@@ -16,44 +16,51 @@ limitations under the License.
 package v1alpha1
 
 import (
+	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// TcpRouteSpec defines the desired state of TcpRoute
-type TcpRouteSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteSpec defines the desired state of the TCPRoute.
+type TCPRouteSpec struct {
+	// Hosts is a list of Host definitions.
+	Hosts []TCPRouteHost `json:"hosts,omitempty"`
 }
 
-// TcpRouteStatus defines the observed state of TcpRoute
-type TcpRouteStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteHost is the configuration for a given host.
+type TCPRouteHost struct {
+	// Hostname is a fully qualified domain name.  This must be a registered
+	// name as defined by RFC 3986.  Note that IP addresses cannot be used.
```

Comment:
Since `Hostname` is sort of overloaded in multiple layers of the OSI model, can we be explicit here that `host` in this case means the `SNI` in the `TLS` protocol?

Reply:
Yeah, we can do that.  I think it would be clearest to define range of acceptable values (as the godoc I wrote attempts to do), and then explain how the value is intended to be used (namely, to match the TLS SNI).  Does that seem reasonable?

Reply:
Sounds good to me. I'm not sure where it should be present, in godoc or somewhere else. At the moment, godoc seems like a good place.